renku/service/entrypoint.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Renku service entry point."""
+import os
+import uuid
+
+from flask import Flask
+from flask_swagger_ui import get_swaggerui_blueprint
+
+from renku.service.config import API_URL, API_VERSION, CACHE_PROJECTS_PATH, \
+    CACHE_UPLOADS_PATH, SWAGGER_URL, UPLOAD_FOLDER
+from renku.service.views.cache import clone_repository_view, \
+    list_projects_view, list_uploaded_files_view, upload_files_view
+from renku.service.views.datasets import add_file_to_dataset_view, \
+    create_dataset_view, list_dataset_files_view, list_datasets_view
+from renku.service.views.docs import api_docs_view
+
+
+def make_cache():
+    """Create cache structure."""
+    sub_dirs = [CACHE_UPLOADS_PATH, CACHE_PROJECTS_PATH]
+
+    for subdir in sub_dirs:
+        if not subdir.exists():
+            subdir.mkdir()
+
+
+def create_app():
+    """Creates a Flask app with necessary configuration."""
+    app = Flask(__name__)
+    app.secret_key = os.getenv('RENKU_SVC_SERVICE_KEY', uuid.uuid4().hex)
+
+    app.config['UPLOAD_FOLDER'] = UPLOAD_FOLDER.name
+    app.config['MAX_CONTENT_LENGTH'] = 16 * 1024 * 1024
+
+    make_cache()
+    build_routes(app)
+
+    return app
+
+
+def build_routes(app):
```

Comment:
These should all be blueprints?

Reply:
As discussed there is a reason for not opting to blueprints immediately.